src/VirtualDeskManager.cpp

```diff
@@ -280,16 +280,16 @@ void VirtualDeskManager::invalidateAllLayouts() {
     }
 }
 
-CMonitor* VirtualDeskManager::getCurrentMonitor() {
-    CMonitor* currentMonitor = g_pCompositor->m_pLastMonitor.get();
+CSharedPointer<CMonitor> VirtualDeskManager::getCurrentMonitor() {
+    CWeakPointer<CMonitor> currentMonitor = g_pCompositor->m_pLastMonitor;
     // This can happen when we receive the "on disconnect" signal
     // let's just take first monitor we can find
     if (currentMonitor && (!currentMonitor->m_bEnabled || !currentMonitor->output)) {
         for (auto mon : g_pCompositor->m_vMonitors) {
             if (mon->m_bEnabled && mon->output)
-                return mon.get();
+                return mon;
         }
         return nullptr;
     }
-    return currentMonitor;
+    return currentMonitor.lock();
```

Comment:
what does this do btw?

Reply:
it's used to convert the weak pointer returned by the hyprland api to a shared pointer